### PR TITLE
Login screen page

### PR DESCRIPTION
**Pull request by Matheus-Du**
Basic UI page for user login along with the corresponding account creation fragment.

All buttons are functional for navigation to/from fragment and can be used to redirect to other pages as necessary.
Pressing the "sign-in" button currently does nothing, but function is present to allow for redirection to the main page.
Pressing the "Register" button will open the account creation fragment, which presents the user with all necessary text fields to create a new account. Both "Sign up" and "cancel" buttons in the fragment close the fragment, but can easily be changed to begin more complex tasks such as account creation.

The only UI element not present is the Logo on the main sign-in page to be added later.

### CODE REVIEW

File: CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/SignUpFragment.java [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+/**
+ * A simple {@link Fragment} subclass.
+ * Use the {@link SignUpFragment#newInstance} factory method to
+ * create an instance of this fragment.
+ */
+public class SignUpFragment extends Fragment  implements View.OnClickListener {
+
+    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
+    private static final String ARG_PARAM1 = "param1";
```
Comment (Ryden-Graham) [inline CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/SignUpFragment.java:33]: I'm not sure what "param1" and "param2" are. Maybe username/password? Can we rename these to something more descriptive?

File: CMPUT_301_Project/app/src/main/res/layout/fragment_sign_up.xml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+                android:layout_marginTop="32dp"
+                android:orientation="horizontal">
+
+                <TextView
+                    android:id="@+id/reenterPasswordTV"
+                    android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:layout_weight="1"
+                    android:lines="2"
+                    android:text="Re-Enter Password:"
```
Comment (Ryden-Graham) [inline CMPUT_301_Project/app/src/main/res/layout/fragment_sign_up.xml:118]: Can we put this in a layout and stack it over two lines? It looks very small at the moment.

Comment (adiep2) [reply CMPUT_301_Project/app/src/main/res/layout/fragment_sign_up.xml:118]: I agree, the text on "Re-Enter Password:" looks small

File: CMPUT_301_Project/app/src/main/res/layout/activity_login_screen_page.xml [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                android:layout_width="match_parent"
+                android:layout_height="48dp"
+                android:layout_weight="1"
+                android:ems="10"
+                android:inputType="textPassword" />
+
+        </LinearLayout>
+
+        <TextView
+            android:id="@+id/textView3"
```
Comment (adiep2) [inline CMPUT_301_Project/app/src/main/res/layout/activity_login_screen_page.xml:86]: Change id name to something more descriptive

File: CMPUT_301_Project/app/src/main/res/layout/activity_login_screen_page.xml [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+            android:layout_height="wrap_content"
+            android:paddingTop="24dp"
+            android:text="Don't have an account yet? Register!"
+            android:textAlignment="center"
+            android:textColor="#000000" />
+
+    </LinearLayout>
+
+    <LinearLayout
+        android:id="@+id/linearLayout3"
```
Comment (adiep2) [inline CMPUT_301_Project/app/src/main/res/layout/activity_login_screen_page.xml:97]: Change id name to something more descriptive

File: CMPUT_301_Project/app/src/main/res/layout/fragment_sign_up.xml [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+                    android:layout_height="wrap_content"
+                    android:layout_weight="32"
+                    android:ems="10"
+                    android:inputType="textPassword" />
+            </LinearLayout>
+
+        </LinearLayout>
+
+        <LinearLayout
+            android:id="@+id/linearLayout2"
```
Comment (adiep2) [inline CMPUT_301_Project/app/src/main/res/layout/fragment_sign_up.xml:133]: Change id name to something more descriptive

**Review by adiep2 — CHANGES_REQUESTED** [review]
Left some comments about minor things that need to be changed (variable names and font sizing).  Overall looks good! Just look over the minor changes and I will approve the PR.

File: CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/SignUpFragment.java (hunk 1 above)
Comment (Matheus-Du) [reply CMPUT_301_Project/app/src/main/java/com/example/cmput_301_project/SignUpFragment.java]: params created when default fragment was created and serve no purpose atm. Removed in newest commit.

File: CMPUT_301_Project/app/src/main/res/layout/fragment_sign_up.xml (hunk 2 above)
Comment (Matheus-Du) [reply CMPUT_301_Project/app/src/main/res/layout/fragment_sign_up.xml]: Problem was with layout widths not working with chosen text size for textviews. Changed all sizes of textviews to match at a slightly smaller size and now the layouts all lineup. 're-enter password' now spans 2 lines and this should be resolved.

File: CMPUT_301_Project/app/src/main/res/layout/activity_login_screen_page.xml (hunk 3 above)
Comment (Matheus-Du) [reply CMPUT_301_Project/app/src/main/res/layout/activity_login_screen_page.xml]: resolved

File: CMPUT_301_Project/app/src/main/res/layout/activity_login_screen_page.xml (hunk 4 above)
Comment (Matheus-Du) [reply CMPUT_301_Project/app/src/main/res/layout/activity_login_screen_page.xml]: resolved

File: CMPUT_301_Project/app/src/main/res/layout/fragment_sign_up.xml (hunk 5 above)
Comment (Matheus-Du) [reply CMPUT_301_Project/app/src/main/res/layout/fragment_sign_up.xml]: resolved

**Review by adiep2 — APPROVED** [review]

**Review by Ryden-Graham — APPROVED** [review]
Checked the font size update, looks good!

**Pull request merged**